### DISCUSSION TITLE
How do I define the resulting file name when running a script loop?

### DISCUSSION THREAD

**Discussion by jordanearth** [ImageMagick/ImageMagick#3752] [Help]
Hey guys. I'm running a script loop based on Fred's multicrop script. I'm on Windows 10 (64-bit). Here's my code. `cd /mnt/c/Users/right/Pictures/Scans/Fred/Images
for img in *tif; do
./multicropx -g 25 -f 20 -u 1 -i yes $img *.tif
done`. I want each resulting file name to be the same name as the source but with a c at the end. For example, Image_34.tif would output Image_34c.tif after running the script.

Bonus question: How do I get the resulting files to go into their own folder separate from the source? All within the same script.

**Comment by fmw42** [comment]
I am not sure what you mean by put in own directory. All output images from a given input image will go to the same (existing) directory. The script will not create directories. Also note that TIFF files can contain multiple images. The script will not process those files. You must select one of the images using the [x] notation. The following will process only the first image in the input TIFF by using [0]. If you want the second one, then use [1]

```
for img in *tif; do
name=`convert $img -format "%t" info:`
./multicropx -g 25 -f 20 -u 1 -i yes $img[0] $path_to_directory/${name}c.tif
done

```

**Reply by whugemann** [reply]
This is rather a question on shell scripting, not on IM usage.

In what shell you are running your script? CMD, Powershell, Cygwin? (See https://legacy.imagemagick.org/Usage/windows/#environments)
